src/main/scala/beam/agentsim/agents/vehicles/BeamVehicle.scala

```diff
@@ -291,12 +291,12 @@ object BeamVehicle {
     beamLeg: BeamLeg,
     theVehicleType: BeamVehicleType,
     networkHelper: NetworkHelper,
-    onlyLength_Id_And_Type: Boolean = false
+    vehicleMappingExists: Boolean = false
```

Comment:
This is not an accurate refactoring of what is meant here. This method was in the process of being pared down to reduce calculating things it didn't need any longer. So, this if else is supposed to indicate that we care to get back `FuelConsumptionData` with ONLY length, id and type filled in. And the only time this happens currently is if the mapping exists. A better rename would be to be even more explicit - `fuelConsumpionDataWithOnlyLength_Id_And_Type`.